tinygrad/runtime/graph/clang.py

```diff
@@ -14,7 +14,8 @@ def __init__(self, jit_cache: List[ExecItem], input_rawbuffers: List[Buffer], va
     super().__init__(jit_cache, input_rawbuffers, var_vals)
     if not all(isinstance(ji.prg, CompiledRunner) for ji in jit_cache): raise GraphException
 
-    prgs = '\n'.join(dedup([cast(CompiledRunner, ji.prg).p.src for ji in jit_cache]))
+    new_decl = "static inline __attribute__((always_inline)) void"
```

Comment:
I don't think this is the right way to deal with graph. Use a linker section to ensure the function runner is always first

Reply:
Ok, sounds good. I think having multiple functions was creating relocations, but I'll see if I can work around this.